src/govuk/components/character-count/character-count.mjs

```diff
@@ -277,22 +318,28 @@ CharacterCount.prototype.count = function (text) {
  * @returns {String} Status message
  */
 CharacterCount.prototype.getCountMessage = function () {
-  var $textarea = this.$textarea
-  var config = this.config
-  var remainingNumber = this.maxLength - this.count($textarea.value)
+  var remainingNumber = this.maxLength - this.count(this.$textarea.value)
 
-  var charVerb = 'remaining'
-  var charNoun = 'character'
-  var displayNumber = remainingNumber
-  if (config.maxwords) {
-    charNoun = 'word'
+  var countType = this.config.maxwords ? 'words' : 'characters'
+  return this.formatCountMessage(remainingNumber, countType)
+}
+
+/**
+ * Formats the message shown to users according to what's counted
+ * and how many remain
+ *
+ * @param {Number} remainingNumber - The number of words/characaters remaining
+ * @param {String} countType - "words" or "characters"
+ * @returns String
+ */
+CharacterCount.prototype.formatCountMessage = function (remainingNumber, countType) {
+  if (remainingNumber === 0) {
+    return this.i18n.t(countType + 'AtLimit')
   }
-  charNoun = charNoun + ((remainingNumber === -1 || remainingNumber === 1) ? '' : 's')
 
-  charVerb = (remainingNumber < 0) ? 'too many' : 'remaining'
-  displayNumber = Math.abs(remainingNumber)
+  var translationKeySuffix = remainingNumber < 0 ? 'OverLimit' : 'UnderLimit'
 
-  return 'You have ' + displayNumber + ' ' + charNoun + ' ' + charVerb
+  return this.i18n.t(countType + translationKeySuffix, { count: Math.abs(remainingNumber) })
```

Comment:
We do `Math.abs(Math.floor(count))` in **i18n.mjs**

Is this enough? Probably can't have decimals can we

Reply:
That's a good catch, but I'd feel safer leaving the `Math.abs` in the CharacterCount:
- it explicits that the CharacterCount wants to display the absolute value in the number, rather than obscuring it by knowing that the I18n does it (which not everyone reading that line will expect)
- it prevents surprises the day I18n expands its support as the line you mention is led by `// Currently our custom code can only handle positive integers`

Reply:
That's fine, just something that jumped out at me 😊